lib/ReadsUtils/ReadsUtilsImpl.py

```diff
@@ -823,15 +846,54 @@ def upload_reads(self, ctx, params):
         self.log('Starting upload reads, parsing args')
         o, wsid, name, objid, kbtype, single_end, fwdid, revid, shock = (
             self._proc_upload_reads_params(params))
-        interleaved = 1 if (not single_end and not revid) else 0
+        #IF from shock fwdid and revid are shock nodes, if not shock they are file paths
+        dfu = DataFileUtil(self.callback_url)
+        fwdpath = None
+        revpath = None
         if shock:
-            fhandle, rhandle, fsize, rsize = self.process_shock_ids_for_upload(
-                fwdid, revid, interleaved)
+            #Grab files from Shock
+            fileinput = [{'shock_id': fwdid,
+                          'file_path': self.scratch + '/fwd/',
+                          'unpack': 'uncompress'}]
+            if revid:
+                fileinput.append({'shock_id': revid,
+                                  'file_path': self.scratch + '/rev/',
+                                  'unpack': 'uncompress'})                            
+            self.log('downloading reads file(s) from Shock')
+            files = dfu.shock_to_file_mass(fileinput)
+            shock = False
+            fwdpath = files[0]["file_path"]
+            fwdname = files[0]["node_file_name"]
+            if revid:
+                revpath = files[1]["file_path"]
+                revname = files[1]["node_file_name"]            
         else:
-            fhandle, rhandle, fsize, rsize = self.process_files_for_upload(
-                fwdid, revid, interleaved)
+            #Not shock
+            fwdpath = fwdid
+            revpath = revid
+            fwdname = None
+            revname = None
+
+        shock = False
+        if revid:
+            #now interleave the files
+            intpath = os.path.join(self.scratch, self.get_file_prefix() +
+                                    '.inter.fastq')
+            self.interleave(
+                    None, None, fwdname, fwdpath,
+                    revname, revpath, fwdpath, revpath, intpath)
+            fwdpath = intpath
+            revid = None
+            revpath = None
+  
+        interleaved = 1 if (not single_end and not revid) else 0        
+
+        fhandle, rhandle, fsize, rsize = self.process_files_for_upload(
+                fwdpath, revpath, interleaved)
```

Comment:
OK, there's actually a bigger problem here - the code now always uses this function to upload file to shock,which assumes that the files were local. That means any error messages from the function don't include the shock IDs from which the files originate, which makes it impossible to debug errors.